### ISSUE TITLE
Data Quality:  Agency Taxonomy (Reference Data Source)

### ISSUE THREAD

**Issue by navarretedaniel** [fedspendingtransparency/USASpending-Issue-Tracker#89]
Apologies ahead of time for the wall of text. 

Is there a definitive source noting the reference data source(s) used for the agency taxonomy(ies) found on USAspending?

Moving forward, I'll use the U.S. Department of Labor, [Wage and Hour Division ](http://www.dol.gov/whd/)as an illustrative example.  Prior to 2009, the Wage and Hour Division was a sub-agency within DOL's Employment Standards Administration (ESA).  Since 2009, it has been a standalone sub-agency (and ESA was dissolved).

In a conversation in Issue #78, a request for clarity concerning agency codes is made.  The response to this request is a link to NIST's April 2008 report "[Codes for Identification of Federal and Federally-Assisted Organizations](http://nvlpubs.nist.gov/nistpubs/Legacy/SP/nistspecialpublication800-87r1.pdf)."

This NIST publication notes:
- Labor, Department of:  1600
- [Employment Standards Administration:  1635]
- Wage and Hour Division:  n/a

Issue #75 poses a similar question.  The response to this issue simply notes that USAspending pulls data submitted to FPDS but it does not provide a direct link to a source file.  

The [Federal Spending Transparency Data Act Collaboration Space](https://fedspendingtransparency.github.io/whitepapers/fundingagency/) provides a link to GSA's [table of agency codes](https://www.fpds.gov/downloads/top_requests/FPDSNG_Contracting_Offices.xls) and office codes as used in the FPDS-NG system: 

This FPDS-NG table notes:
- Labor, Department of:  1600
- [Employment Standards Administration:  n/a]
- Wage and Hour Division:  n/a

GSA has a [blog post](https://interact.gsa.gov/blog/standardized-awarding-and-funding-office-codes-now-required-fpds-ng) dated March 31, 2016, discussing the need to standardize awarding and funding office codes required in FPDS-NG.  This post defines an Activity Address Code as a uniform way to idenfity organizations in federal agencies.  This blog post links to a Max.gov team site tasked with determining these agency codes.  A file within this site lists:
- Labor, Department of:  1600
- Wage and Hour Division:  1631

Similarly, GSA maintains a separate [table of agency and bureau codes](http://www.gsa.gov/graphics/pbs/eRETA_Agency_Bureau_Codes.xls) for purposes of Reimbursable Work Authorizations (RWAs) and its corresponding [RWA Entry and Tracking Application (RETA)](http://www.gsa.gov/portal/category/26813).  
- Department of Labor:  01600
- Wage and Hour Division:  01631

And finally, USAspending's "Data Download" agency filter provides a bureau code of 1636 for the "Wage An Hour Division" (note the typo; happy to open a new issue to correct this misspelling, if needed).
![image](https://cloud.githubusercontent.com/assets/18538131/19567952/b55c81a8-96bd-11e6-983d-c7e753543a32.png)

In short, I'm asking:

1.  Is there a definitive source noting the reference data source(s) used for the agency taxonomy(ies) found on USAspending?
2.  Which source document provides "1636" as the bureau code for the Wage and Hour Division?
3.  Will USAspending be adopting the proposed agency and bureau codes referenced in the March 2016 GSA blog post?
4.  If not, are there plans to change to a new agency taxonomy any time in the future?
5.  If multiple taxonomies are used (i.e., agency codes as reported in FPDS; agency codes found on Data Download page; etc.), is there a crosswalk available?

Thank you!
